src/page/slot.rs

```diff
@@ -266,21 +214,25 @@ impl<T, C: cfg::Config> Slot<T, C> {
         // will have to be removed when those references are dropped.
         if refs.value > 0 {
             return true;
+        } else {
+            false
         }
-
-        // Otherwise, we can remove the slot now!
-        test_println!("-> remove deferred; can remove now");
-        let removed = self.remove_value(curr_gen, offset, free).is_some();
-        removed
     }
 
-    #[inline]
-    pub(super) fn remove_value<F: FreeList<C>>(
+    /// Mutates this slot.
+    ///
+    /// This method spins until no references to this slot are left, and calls the mutator
+    fn mutate_storage_with_confirmation<F, M, R>(
         &self,
         gen: Generation<C>,
         offset: usize,
         free: &F,
-    ) -> Option<T> {
+        mutator: M,
+    ) -> Option<R>
```

Comment:
similarly to what i said in https://github.com/hawkw/sharded-slab/pull/19#discussion_r389066731, i think this should just return `R`
```suggestion
    ) -> R
```
since the method always spins until there are no outstanding refs, the function will always be called _eventually_, so we never return `None`; this only returned `Option` because the function invoked to clear the slot used to _always_ be `Option::take`. in the slab case, where we still use `take`, this will then return `Option<Option<T>>`, where the outer `Option` will always be `Some`.